Failing to use haskell-language-server on itself

Hi,

I searched in the previous issues but couldn't find any related issue.

I do not manage to use haskell-language-server on itself 🤔 Basically I wanted to start hacking/looking at the code, and noticed in VSCode that all hovers get stuck on "Loading...".

Steps to reproduce:
1. `git clone https://github.com/haskell/haskell-language-server --recurse-submodules`
2. `cd haskell-language-server`
3. `stack ./install.hs hls-8.6.5` (`8.6.5` because `stack.yaml` uses `lts-14.27` which relies on GHC 8.6.5)
4. `cp hie.yaml.stack hie.yaml`
5. `haskell-language-server`

I get 3 errors and `haskell-language-server: <command line>: cannot satisfy -package z-haskell-language-server-z-hls-test-utils`.

In VSCode, surprisingly, I don't get the 3 compiler errors, only the last `haskell-language-server: <command line>: cannot satisfy -package z-haskell-language-server-z-hls-test-utils` bit 🤔 Maybe this is related to the fact that VS Code runs it in LSP mode though.

Below the full logs (second run, the first is the same but with a lot of `using precompiled language`):
```
ghcide version: 0.1.0.0 (GHC: 8.6.5) (PATH: /home/sir4ur0n/.local/bin/haskell-language-server) (GIT hash: 6c58c2d2147b14a4fd21f6d95183125d46141f3d)
(haskell-language-server)Ghcide setup tester in /home/sir4ur0n/code/haskell-language-server.
Report bugs at https://github.com/haskell/haskell-language-server/issues

Step 1/6: Finding files to test in /home/sir4ur0n/code/haskell-language-server
Found 201 files

Step 2/6: Looking for hie.yaml files that control setup
Found 2 cradles

Step 3/6: Initializing the IDE

Step 4/6: Type checking the files
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/install/src/Stack.hs"
Left [CradleError ExitSuccess ["Multi Cradle: No prefixes matched","pwd: /home/sir4ur0n/code/haskell-language-server","filepath: /home/sir4ur0n/code/haskell-language-server/install/src/Stack.hs","prefixes:","(\"./test/functional/\",Stack {component = Just \"haskell-language-server:func-test\"})","(\"./test/src/\",Stack {component = Just \"haskell-language-server:tasty\"})","(\"./exe/Main.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server\"})","(\"./exe/Wrapper.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server-wrapper\"})","(\"./src\",Stack {component = Just \"haskell-language-server:lib\"})","(\"./ghcide/src\",Stack {component = Just \"ghcide:lib:ghcide\"})","(\"./ghcide/exe\",Stack {component = Just \"ghcide:exe:ghcide\"})"]]
File:     /home/sir4ur0n/code/haskell-language-server/install/src/Stack.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/A/Main.hs"
File:     /home/sir4ur0n/code/haskell-language-server/install/src/Stack.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Multi Cradle: No prefixes matched
  pwd: /home/sir4ur0n/code/haskell-language-server
Left [CradleError ExitSuccess ["Multi Cradle: No prefixes matched","pwd: /home/sir4ur0n/code/haskell-language-server","filepath: /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/A/Main.hs","prefixes:","(\"./test/functional/\",Stack {component = Just \"haskell-language-server:func-test\"})","(\"./test/src/\",Stack {component = Just \"haskell-language-server:tasty\"})","(\"./exe/Main.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server\"})","(\"./exe/Wrapper.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server-wrapper\"})","(\"./src\",Stack {component = Just \"haskell-language-server:lib\"})","(\"./ghcide/src\",Stack {component = Just \"ghcide:lib:ghcide\"})","(\"./ghcide/exe\",Stack {component = Just \"ghcide:exe:ghcide\"})"]]
File:     /home/sir4ur0n/code/haskell-language-server/install/src/Stack.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Multi Cradle: No prefixes matched
  pwd: /home/sir4ur0n/code/haskell-language-server
  filepath: /home/sir4ur0n/code/haskell-language-server/install/src/Stack.hs
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs"
  prefixes:
  ("./test/functional/",Stack {component = Just "haskell-language-server:func-test"})
  ("./test/src/",Stack {component = Just "haskell-language-server:tasty"})
  ("./exe/Main.hs",Stack {component = Just "haskell-language-server:exe:haskell-language-server"})
  ("./exe/Wrapper.hs",Stack {component = Just
  "haskell-language-server:exe:haskell-language-server-wrapper"})
  ("./src",Stack {component = Just "haskell-language-server:lib"})
  ("./ghcide/src",Stack {component = Just "ghcide:lib:ghcide"})
  ("./ghcide/exe",Stack {component = Just "ghcide:exe:ghcide"})
File:     /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/A/Main.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Multi Cradle: No prefixes matched
  pwd: /home/sir4ur0n/code/haskell-language-server
  filepath:
  /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/A/Main.hs
  prefixes:
  ("./test/functional/",Stack {component = Just "haskell-language-server:func-test"})
  ("./test/src/",Stack {component = Just "haskell-language-server:tasty"})
  ("./exe/Main.hs",Stack {component = Just "haskell-language-server:exe:haskell-language-server"})
  ("./exe/Wrapper.hs",Stack {component = Just
  "haskell-language-server:exe:haskell-language-server-wrapper"})
  ("./src",Stack {component = Just "haskell-language-server:lib"})
  ("./ghcide/src",Stack {component = Just "ghcide:lib:ghcide"})
  ("./ghcide/exe",Stack {component = Just "ghcide:exe:ghcide"})
> Cloning into '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-cfd5b37729d07439'...
> HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.
> Resolving dependencies...
> cabal: Could not resolve dependencies:
> [__0] trying: ghcide-0.1.0 (user goal)
> [__1] trying: ghcide:-ghc-lib
> [__2] trying: hie-bios-0.5.0 (dependency of ghcide -ghc-lib)
> [__3] trying: yaml-0.11.4.0 (dependency of hie-bios)
> [__4] trying: template-haskell-2.11.1.0/installed-2.1... (dependency of yaml)
> [__5] next goal: pretty (dependency of template-haskell)
> [__5] rejecting: pretty-1.1.3.3/installed-1.1... (conflict: hie-bios =>
> deepseq>=1.4.3 && <1.5, pretty => deepseq==1.4.2.0/installed-1.4...)
> [__5] rejecting: pretty-1.1.3.6, pretty-1.1.3.5, pretty-1.1.3.4,
> pretty-1.1.3.3, pretty-1.1.3.2, pretty-1.1.3.1, pretty-1.1.2.1,
> pretty-1.1.2.0, pretty-1.1.1.3, pretty-1.1.1.2, pretty-1.1.1.1,
> pretty-1.1.1.0, pretty-1.1.0.0, pretty-1.0.1.2, pretty-1.0.1.1,
> pretty-1.0.1.0, pretty-1.0.0.0 (conflict: template-haskell =>
> pretty==1.1.3.3/installed-1.1...)
> [__5] fail (backjumping, conflict set: hie-bios, pretty, template-haskell)
> After searching the rest of the dependency tree exhaustively, these were the
> goals I've had most trouble fulfilling: template-haskell, pretty, hie-bios,
> yaml, ghcide, ghcide:ghc-lib
> Try running with --minimize-conflict-set to improve the error message.
>
Left [CradleError (ExitFailure 1) ["Failed to parse result of calling cabal","Resolving dependencies...\n","Cloning into '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-cfd5b37729d07439'...\nHEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.\ncabal: Could not resolve dependencies:\n[__0] trying: ghcide-0.1.0 (user goal)\n[__1] trying: ghcide:-ghc-lib\n[__2] trying: hie-bios-0.5.0 (dependency of ghcide -ghc-lib)\n[__3] trying: yaml-0.11.4.0 (dependency of hie-bios)\n[__4] trying: template-haskell-2.11.1.0/installed-2.1... (dependency of yaml)\n[__5] next goal: pretty (dependency of template-haskell)\n[__5] rejecting: pretty-1.1.3.3/installed-1.1... (conflict: hie-bios =>\ndeepseq>=1.4.3 && <1.5, pretty => deepseq==1.4.2.0/installed-1.4...)\n[__5] rejecting: pretty-1.1.3.6, pretty-1.1.3.5, pretty-1.1.3.4,\npretty-1.1.3.3, pretty-1.1.3.2, pretty-1.1.3.1, pretty-1.1.2.1,\npretty-1.1.2.0, pretty-1.1.1.3, pretty-1.1.1.2, pretty-1.1.1.1,\npretty-1.1.1.0, pretty-1.1.0.0, pretty-1.0.1.2, pretty-1.0.1.1,\npretty-1.0.1.0, pretty-1.0.0.0 (conflict: template-haskell =>\npretty==1.1.3.3/installed-1.1...)\n[__5] fail (backjumping, conflict set: hie-bios, pretty, template-haskell)\nAfter searching the rest of the dependency tree exhaustively, these were the\ngoals I've had most trouble fulfilling: template-haskell, pretty, hie-bios,\nyaml, ghcide, ghcide:ghc-lib\nTry running with --minimize-conflict-set to improve the error message.\n\n",""]]
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs
Hidden:   no
Range:    1:0-100001:0
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs"
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Failed to parse result of calling cabal
  Resolving dependencies...

  Cloning into
  '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-cfd5b37729d07439'...
  HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to
  guide scheduling of short actions on a single core.
  cabal: Could not resolve dependencies:
  [__0] trying: ghcide-0.1.0 (user goal)
  [__1] trying: ghcide:-ghc-lib
  [__2] trying: hie-bios-0.5.0 (dependency of ghcide -ghc-lib)
  [__3] trying: yaml-0.11.4.0 (dependency of hie-bios)
  [__4] trying: template-haskell-2.11.1.0/installed-2.1... (dependency of yaml)
  [__5] next goal: pretty (dependency of template-haskell)
  [__5] rejecting: pretty-1.1.3.3/installed-1.1... (conflict: hie-bios =>
  deepseq>=1.4.3 && <1.5, pretty => deepseq==1.4.2.0/installed-1.4...)
  [__5] rejecting: pretty-1.1.3.6, pretty-1.1.3.5, pretty-1.1.3.4,
  pretty-1.1.3.3, pretty-1.1.3.2, pretty-1.1.3.1, pretty-1.1.2.1,
  pretty-1.1.2.0, pretty-1.1.1.3, pretty-1.1.1.2, pretty-1.1.1.1,
  pretty-1.1.1.0, pretty-1.1.0.0, pretty-1.0.1.2, pretty-1.0.1.1,
  pretty-1.0.1.0, pretty-1.0.0.0 (conflict: template-haskell =>
  pretty==1.1.3.3/installed-1.1...)
  [__5] fail (backjumping, conflict set: hie-bios, pretty, template-haskell)
  After searching the rest of the dependency tree exhaustively, these were the
  goals I've had most trouble fulfilling: template-haskell, pretty, hie-bios,
  yaml, ghcide, ghcide:ghc-lib
  Try running with --minimize-conflict-set to improve the error message.



> HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.
> Resolving dependencies...
> cabal: Could not resolve dependencies:
> [__0] trying: ghcide-0.1.0 (user goal)
> [__1] trying: ghcide:-ghc-lib
> [__2] trying: hie-bios-0.5.0 (dependency of ghcide -ghc-lib)
> [__3] trying: yaml-0.11.4.0 (dependency of hie-bios)
> [__4] trying: template-haskell-2.11.1.0/installed-2.1... (dependency of yaml)
> [__5] next goal: pretty (dependency of template-haskell)
> [__5] rejecting: pretty-1.1.3.3/installed-1.1... (conflict: hie-bios =>
> deepseq>=1.4.3 && <1.5, pretty => deepseq==1.4.2.0/installed-1.4...)
> [__5] rejecting: pretty-1.1.3.6, pretty-1.1.3.5, pretty-1.1.3.4,
> pretty-1.1.3.3, pretty-1.1.3.2, pretty-1.1.3.1, pretty-1.1.2.1,
> pretty-1.1.2.0, pretty-1.1.1.3, pretty-1.1.1.2, pretty-1.1.1.1,
> pretty-1.1.1.0, pretty-1.1.0.0, pretty-1.0.1.2, pretty-1.0.1.1,
> pretty-1.0.1.0, pretty-1.0.0.0 (conflict: template-haskell =>
> pretty==1.1.3.3/installed-1.1...)
> [__5] fail (backjumping, conflict set: hie-bios, pretty, template-haskell)
> After searching the rest of the dependency tree exhaustively, these were the
> goals I've had most trouble fulfilling: template-haskell, pretty, hie-bios,
> yaml, ghcide, ghcide:ghc-lib
> Try running with --minimize-conflict-set to improve the error message.
>
Left [CradleError (ExitFailure 1) ["Failed to parse result of calling cabal","Resolving dependencies...\n","HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.\ncabal: Could not resolve dependencies:\n[__0] trying: ghcide-0.1.0 (user goal)\n[__1] trying: ghcide:-ghc-lib\n[__2] trying: hie-bios-0.5.0 (dependency of ghcide -ghc-lib)\n[__3] trying: yaml-0.11.4.0 (dependency of hie-bios)\n[__4] trying: template-haskell-2.11.1.0/installed-2.1... (dependency of yaml)\n[__5] next goal: pretty (dependency of template-haskell)\n[__5] rejecting: pretty-1.1.3.3/installed-1.1... (conflict: hie-bios =>\ndeepseq>=1.4.3 && <1.5, pretty => deepseq==1.4.2.0/installed-1.4...)\n[__5] rejecting: pretty-1.1.3.6, pretty-1.1.3.5, pretty-1.1.3.4,\npretty-1.1.3.3, pretty-1.1.3.2, pretty-1.1.3.1, pretty-1.1.2.1,\npretty-1.1.2.0, pretty-1.1.1.3, pretty-1.1.1.2, pretty-1.1.1.1,\npretty-1.1.1.0, pretty-1.1.0.0, pretty-1.0.1.2, pretty-1.0.1.1,\npretty-1.0.1.0, pretty-1.0.0.0 (conflict: template-haskell =>\npretty==1.1.3.3/installed-1.1...)\n[__5] fail (backjumping, conflict set: hie-bios, pretty, template-haskell)\nAfter searching the rest of the dependency tree exhaustively, these were the\ngoals I've had most trouble fulfilling: template-haskell, pretty, hie-bios,\nyaml, ghcide, ghcide:ghc-lib\nTry running with --minimize-conflict-set to improve the error message.\n\n",""]]
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/src/Ide/Logger.hs"
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Failed to parse result of calling cabal
  Resolving dependencies...

  HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to
  guide scheduling of short actions on a single core.
  cabal: Could not resolve dependencies:
  [__0] trying: ghcide-0.1.0 (user goal)
  [__1] trying: ghcide:-ghc-lib
  [__2] trying: hie-bios-0.5.0 (dependency of ghcide -ghc-lib)
  [__3] trying: yaml-0.11.4.0 (dependency of hie-bios)
  [__4] trying: template-haskell-2.11.1.0/installed-2.1... (dependency of yaml)
  [__5] next goal: pretty (dependency of template-haskell)
  [__5] rejecting: pretty-1.1.3.3/installed-1.1... (conflict: hie-bios =>
  deepseq>=1.4.3 && <1.5, pretty => deepseq==1.4.2.0/installed-1.4...)
  [__5] rejecting: pretty-1.1.3.6, pretty-1.1.3.5, pretty-1.1.3.4,
  pretty-1.1.3.3, pretty-1.1.3.2, pretty-1.1.3.1, pretty-1.1.2.1,
  pretty-1.1.2.0, pretty-1.1.1.3, pretty-1.1.1.2, pretty-1.1.1.1,
  pretty-1.1.1.0, pretty-1.1.0.0, pretty-1.0.1.2, pretty-1.0.1.1,
  pretty-1.0.1.0, pretty-1.0.0.0 (conflict: template-haskell =>
  pretty==1.1.3.3/installed-1.1...)
  [__5] fail (backjumping, conflict set: hie-bios, pretty, template-haskell)
  After searching the rest of the dependency tree exhaustively, these were the
  goals I've had most trouble fulfilling: template-haskell, pretty, hie-bios,
  yaml, ghcide, ghcide:ghc-lib
  Try running with --minimize-conflict-set to improve the error message.



> haskell-language-server> configure (lib + internal-lib + exe)
> Configuring haskell-language-server-0.1.0.0...
> haskell-language-server> initial-build-steps (lib + internal-lib + exe)
> Configuring GHCi with the following packages: haskell-language-server
> /home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux/4242af1e5fb216b0d2a47f3b5a44999e35ce098c7a4033235b9253d656a0b839/8.6.5/pkgdb:/home/sir4ur0n/.stack/snapshots/x86_64-linux/4242af1e5fb216b0d2a47f3b5a44999e35ce098c7a4033235b9253d656a0b839/8.6.5/pkgdb:/home/sir4ur0n/.stack/programs/x86_64-linux/ghc-8.6.5/lib/ghc-8.6.5/package.conf.d
Right (ComponentOptions {componentOptions = ["-i","-odir=/home/sir4ur0n/code/haskell-language-server/.stack-work/odir","-hidir=/home/sir4ur0n/code/haskell-language-server/.stack-work/odir","-hide-all-packages","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build","-i/home/sir4ur0n/code/haskell-language-server/src","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/autogen","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build/global-autogen","-stubdir=/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux/Cabal-2.4.0.1/build","-package=z-haskell-language-server-z-hls-test-utils","-package-id=base-4.12.0.0","-package-id=aeson-1.4.6.0-4uBfdXbzscjDeNqrTRfhWo","-package-id=binary-0.8.6.0","-package-id=bytestring-0.10.8.2","-package-id=Cabal-3.0.2.0-3jfu38O1DeD8P2u1zPtBbC","-package-id=cabal-helper-1.1.0.0-DrGwajU6uQBJXzpeDNViCO","-package-id=containers-0.6.0.1","-package-id=data-default-0.7.1.1-COovZVyOTYqEavTGLlfqy8","-package-id=deepseq-1.4.4.0","-package-id=Diff-0.3.4-C6Bl1un4oiiBWGaHLlh5bu","-package-id=directory-1.3.3.0","-package-id=extra-1.7.1-DnfiWqwxEy2OHWHxdHnCT","-package-id=filepath-1.4.2.1","-package-id=floskell-0.10.2-EMrS0imTu9T9A3ifnigzE2","-package-id=ghc-8.6.5","-package-id=ghcide-0.1.0-BeX44vabl7jLy1E9mAxkI9","-package-id=gitrev-1.3.1-8kiHdfyRMvw1nWlvJj9ecZ","-package-id=hashable-1.2.7.0-2SI038axTEd7AEZJ275kpi","-package-id=haskell-lsp-0.22.0.0-Dm3Y5tmF7yh4ECsA5etaU6","-package-id=hie-bios-0.5.0-FSBb2ut8sK4IXHsh2geoHJ","-package-id=hslogger-1.2.12-EOh2s68fQfBFyuAt5sDs18","-package-id=lens-4.18-HbPjKqYp8F3CoEHux0jQ2","-package-id=ormolu-0.0.5.0-KfthQtMuKHrDkm8OPx7ZH2","-package-id=optparse-simple-0.1.1.2-HSIl6Lj5MIz5lTvuDIhitS","-package-id=process-1.6.5.0","-package-id=regex-tdfa-1.3.1.0-LDROcq679OH5OnpszsUxo","-package-id=shake-0.19-Cn9Fh99JVyzEo9h0pJqFaB","-package-id=text-1.2.3.1","-package-id=transformers-0.5.6.2","-package-id=unordered-containers-0.2.10.0-LgoTL3wbBEY5bZIDJiyxW4","-package-id=unix-2.7.2.2","-package-id=brittany-0.12.1.1-4VsYnoWOShhGsdS0XyT3CY","-Wall","-Wredundant-constraints","-Wno-name-shadowing","-DAGPL","-optP-include","-optP/home/sir4ur0n/code/haskell-language-server/.stack-work/ghci/73152f3c/cabal_macros.h","-ghci-script=/tmp/haskell-stack-ghci/85f25946/ghci-script","-package-db","/home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux/4242af1e5fb216b0d2a47f3b5a44999e35ce098c7a4033235b9253d656a0b839/8.6.5/pkgdb","-package-db","/home/sir4ur0n/.stack/snapshots/x86_64-linux/4242af1e5fb216b0d2a47f3b5a44999e35ce098c7a4033235b9253d656a0b839/8.6.5/pkgdb","-package-db","/home/sir4ur0n/.stack/programs/x86_64-linux/ghc-8.6.5/lib/ghc-8.6.5/package.conf.d"], componentRoot = "/home/sir4ur0n/code/haskell-language-server", componentDependencies = ["haskell-language-server.cabal","package.yaml","stack.yaml"]})
haskell-language-server: <command line>: cannot satisfy -package z-haskell-language-server-z-hls-test-utils
    (use -v for more information)
```

For what it's worth:
* Windows WSL (Ubuntu 18.04 LTS)
* I have correctly run `sudo apt install libicu-dev libtinfo-dev libgmp-dev`
* Stack version Version 2.3.1, Git revision de2a7b694f07de7e6cf17f8c92338c16286b2878 (8103 commits) x86_64 hpack-0.33.0
* VS Code (using the WSL plugin) 1.45.1
* In the Language Server Haskell (VS Code) plugin, I have changed the following:
  * Formatting provider: ormolu
  * Hie variant: haskell-language-server

I just tested again and the problem seems to persist 🤔 I started again from a fresh clone as stated in my OP.

# First run

## Errors
* `fdLock: invalid argument (Invalid argument)` for `/home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs`
* `fdLock: invalid argument (Invalid argument)` for `/home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs`
* `haskell-language-server: <command line>: cannot satisfy -package haskell-language-server-0.1.0.0` though this seems to be a warning? I'm not sure

## Full logs
```
ghcide version: 0.1.0.0 (GHC: 8.6.5) (PATH: /home/sir4ur0n/.local/bin/haskell-language-server) (GIT hash: 2310e1039c90d9bccaffd688a0d1ec79a9360bf4)
(haskell-language-server)Ghcide setup tester in /home/sir4ur0n/code/haskell-language-server.
Report bugs at https://github.com/haskell/haskell-language-server/issues

Step 1/6: Finding files to test in /home/sir4ur0n/code/haskell-language-server
Found 201 files

Step 2/6: Looking for hie.yaml files that control setup
Found 2 cradles

Step 3/6: Initializing the IDE

Step 4/6: Type checking the files
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs"
> Cloning into '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-5f144bd86bc7c753'...
> HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.
> fdLock: invalid argument (Invalid argument)
Left [CradleError (ExitFailure 1) ["Failed to parse result of calling cabal","","Cloning into '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-5f144bd86bc7c753'...\nHEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.\nfdLock: invalid argument (Invalid argument)\n",""]]
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Failed to parse result of calling cabal

  Cloning into
  '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-5f144bd86bc7c753'...
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs"
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/test/exe/Main.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Failed to parse result of calling cabal

  Cloning into
  '/home/sir4ur0n/code/haskell-language-server/ghcide/dist-newstyle/src/shake-5f144bd86bc7c753'...
  HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to
  guide scheduling of short actions on a single core.
  fdLock: invalid argument (Invalid argument)


> HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.
> fdLock: invalid argument (Invalid argument)
Left [CradleError (ExitFailure 1) ["Failed to parse result of calling cabal","","HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to guide scheduling of short actions on a single core.\nfdLock: invalid argument (Invalid argument)\n",""]]
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs
Hidden:   no
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/test/functional/Diagnostic.hs"
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/src-ghc88/Development/IDE/GHC/HieBin.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Failed to parse result of calling cabal

  HEAD is now at fb3859dc Remove shake scheduler, and use previous action run times as estimates to
  guide scheduling of short actions on a single core.
  fdLock: invalid argument (Invalid argument)


> Using main module: 1. Package `haskell-language-server' component haskell-language-server:test:func-test with main-is file: /home/sir4ur0n/code/haskell-language-server/test/functional/Main.hs
> ghcide                 > configure (lib + exe)
> ghcide                 > Configuring ghcide-0.1.0...
> ghcide                 > build (lib + exe)
> ghcide                 > Preprocessing executable 'ghcide-test-preprocessor' for ghcide-0.1.0..
> ghcide                 > Building executable 'ghcide-test-preprocessor' for ghcide-0.1.0..
> ghcide                 > Preprocessing library for ghcide-0.1.0..
> ghcide                 > Building library for ghcide-0.1.0..
> ghcide                 > Preprocessing executable 'ghcide' for ghcide-0.1.0..
> ghcide                 > Building executable 'ghcide' for ghcide-0.1.0..
> ghcide                 > [3 of 4] Compiling Paths_ghcide
> ghcide                 > [4 of 4] Compiling Main [TH]
> ghcide                 >
> ghcide                 > /home/sir4ur0n/code/haskell-language-server/ghcide/exe/Main.hs:56:1: warning: [-Wunused-imports]
> ghcide                 >     The import of ‘Development.GitRev’ is redundant
> ghcide                 >       except perhaps to import instances from ‘Development.GitRev’
> ghcide                 >     To import instances alone, use: import Development.GitRev()
> ghcide                 >    |
> ghcide                 > 56 | import Development.GitRev
> ghcide                 >    | ^^^^^^^^^^^^^^^^^^^^^^^^^
> ghcide                 >
> ghcide                 > /home/sir4ur0n/code/haskell-language-server/ghcide/exe/Main.hs:57:1: warning: [-Wunused-imports]
> ghcide                 >     The import of ‘action’ from module ‘Development.Shake’ is redundant
> ghcide                 >    |
> ghcide                 > 57 | import Development.Shake (Action,  action)
> ghcide                 >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> ghcide                 >
> ghcide                 > /home/sir4ur0n/code/haskell-language-server/ghcide/exe/Main.hs:83:1: warning: [-Wunused-imports]
> ghcide                 >     The import of ‘Debug.Trace’ is redundant
> ghcide                 >       except perhaps to import instances from ‘Debug.Trace’
> ghcide                 >     To import instances alone, use: import Debug.Trace()
> ghcide                 >    |
> ghcide                 > 83 | import Debug.Trace
> ghcide                 >    | ^^^^^^^^^^^^^^^^^^
> ghcide                 > Linking .stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/ghcide/ghcide ...
> ghcide                 > copy/register
> ghcide                 > Installing executable ghcide-test-preprocessor in /home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/bin
> ghcide                 > Installing library in /home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/lib/x86_64-linux-ghc-8.6.5/ghcide-0.1.0-BeX44vabl7jLy1E9mAxkI9
> ghcide                 > Installing executable ghcide in /home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/bin
> ghcide                 > Registering library for ghcide-0.1.0..
> haskell-language-server> configure (lib + exe)
> Configuring haskell-language-server-0.1.0.0...
> haskell-language-server> initial-build-steps (lib + exe)
> Completed 2 action(s).
> The following GHC options are incompatible with GHCi and have not been passed to it: -threaded
> Configuring GHCi with the following packages: haskell-language-server
>
> Warning: Didn't find expected autogen file:
>          /home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test/autogen/cabal_macros.h
> /home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb:/home/sir4ur0n/.stack/snapshots/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb:/home/sir4ur0n/.stack/programs/x86_64-linux/ghc-tinfo6-8.6.5/lib/ghc-8.6.5/package.conf.d
Right (ComponentOptions {componentOptions = ["-i","-odir=/home/sir4ur0n/code/haskell-language-server/.stack-work/odir","-hidir=/home/sir4ur0n/code/haskell-language-server/.stack-work/odir","-hide-all-packages","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test","-i/home/sir4ur0n/code/haskell-language-server/test/utils","-i/home/sir4ur0n/code/haskell-language-server/test/functional","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test/autogen","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/global-autogen","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test/func-test-tmp","-stubdir=/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build","-package-id=base-4.12.0.0","-package=haskell-language-server-0.1.0.0","-package-id=haskell-lsp-0.22.0.0-Dm3Y5tmF7yh4ECsA5etaU6","-package-id=hie-bios-0.5.0-FSBb2ut8sK4IXHsh2geoHJ","-package-id=aeson-1.4.6.0-4uBfdXbzscjDeNqrTRfhWo","-package-id=blaze-markup-0.8.2.3-ATcS0IRQcDMCxhB8vW9N1d","-package-id=containers-0.6.0.1","-package-id=data-default-0.7.1.1-COovZVyOTYqEavTGLlfqy8","-package-id=directory-1.3.3.0","-package-id=filepath-1.4.2.1","-package-id=hslogger-1.2.12-EOh2s68fQfBFyuAt5sDs18","-package-id=hspec-2.7.1-28Bzukly5w86FpiwMyuAYC","-package-id=hspec-core-2.7.1-IbyPArxIL6526nHHCZLmW2","-package-id=lsp-test-0.10.3.0-8UmPD9MvFJRH3G2kYk9Unq","-package-id=stm-2.5.0.0","-package-id=tasty-hunit-0.10.0.2-FgUrIbgCsUcDDS8XJ0iyJm","-package-id=text-1.2.3.1","-package-id=unordered-containers-0.2.10.0-LgoTL3wbBEY5bZIDJiyxW4","-package-id=yaml-0.11.2.0-8COcGy2e3i5ABQkqGF0U1e","-package-id=haskell-lsp-types-0.22.0.0-EqCdNjFLrl1H6FLGsAPz20","-package-id=hspec-expectations-0.8.2-xEhsaJyqHC4OUn6v2CIJJ","-package-id=lens-4.18-HbPjKqYp8F3CoEHux0jQ2","-package-id=tasty-1.2.3-GiudK7yenNPC8Gtev28Pwr","-package=tasty-ant-xml-1.1.6","-package=tasty-expected-failure-0.11.1.2","-package=tasty-rerun-1.1.17","-Wall","-Wredundant-constraints","-Wall","-Wno-name-shadowing","-rtsopts","-with-rtsopts=-N","-DAGPL","-DAGPL","-ghci-script=/tmp/haskell-stack-ghci/e65fba7d/ghci-script","-package-db","/home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb","-package-db","/home/sir4ur0n/.stack/snapshots/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb","-package-db","/home/sir4ur0n/.stack/programs/x86_64-linux/ghc-tinfo6-8.6.5/lib/ghc-8.6.5/package.conf.d"], componentRoot = "/home/sir4ur0n/code/haskell-language-server", componentDependencies = ["haskell-language-server.cabal","package.yaml","stack.yaml"]})
haskell-language-server: <command line>: cannot satisfy -package haskell-language-server-0.1.0.0
    (use -v for more information)
```

# Second run

## Errors
* `Multi Cradle: No prefixes matched` for `/home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/B/MyLib.hs`
* `Multi Cradle: No prefixes matched` for `/home/sir4ur0n/code/haskell-language-server/test/testdata/HaReRename.hs`
* `haskell-language-server: <command line>: cannot satisfy -package haskell-language-server-0.1.0.0`

## Full logs
```
ghcide version: 0.1.0.0 (GHC: 8.6.5) (PATH: /home/sir4ur0n/.local/bin/haskell-language-server) (GIT hash: 2310e1039c90d9bccaffd688a0d1ec79a9360bf4)
(haskell-language-server)Ghcide setup tester in /home/sir4ur0n/code/haskell-language-server.
Report bugs at https://github.com/haskell/haskell-language-server/issues

Step 1/6: Finding files to test in /home/sir4ur0n/code/haskell-language-server
Found 201 files

Step 2/6: Looking for hie.yaml files that control setup
Found 2 cradles

Step 3/6: Initializing the IDE

Step 4/6: Type checking the files
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/B/MyLib.hs"
Left [CradleError ExitSuccess ["Multi Cradle: No prefixes matched","pwd: /home/sir4ur0n/code/haskell-language-server","filepath: /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/B/MyLib.hs","prefixes:","(\"./test/functional/\",Stack {component = Just \"haskell-language-server:func-test\"})","(\"./test/utils/\",Stack {component = Just \"haskell-language-server:func-test\"})","(\"./exe/Main.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server\"})","(\"./exe/Wrapper.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server-wrapper\"})","(\"./src\",Stack {component = Just \"haskell-language-server:lib\"})","(\"./ghcide/src\",Stack {component = Just \"ghcide:lib:ghcide\"})","(\"./ghcide/exe\",Stack {component = Just \"ghcide:exe:ghcide\"})"]]
File:
  /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/B/MyLib.hs
Hidden:   no
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/ghcide/src/Development/IDE/Core/OfInterest.hs"
File:
  /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/B/MyLib.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Multi Cradle: No prefixes matched
  pwd: /home/sir4ur0n/code/haskell-language-server
  filepath:
  /home/sir4ur0n/code/haskell-language-server/test/testdata/cabal-helper/mono-repo/B/MyLib.hs
  prefixes:
  ("./test/functional/",Stack {component = Just "haskell-language-server:func-test"})
  ("./test/utils/",Stack {component = Just "haskell-language-server:func-test"})
  ("./exe/Main.hs",Stack {component = Just "haskell-language-server:exe:haskell-language-server"})
  ("./exe/Wrapper.hs",Stack {component = Just
  "haskell-language-server:exe:haskell-language-server-wrapper"})
  ("./src",Stack {component = Just "haskell-language-server:lib"})
  ("./ghcide/src",Stack {component = Just "ghcide:lib:ghcide"})
  ("./ghcide/exe",Stack {component = Just "ghcide:exe:ghcide"})
> fdLock: invalid argument (Invalid argument)
Left [CradleError (ExitFailure 1) ["Failed to parse result of calling cabal","","fdLock: invalid argument (Invalid argument)\n",""]]
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/src/Development/IDE/Core/OfInterest.hs
Hidden:   no
Range:    1:0-100001:0
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/test/testdata/HaReRename.hs"
File:     /home/sir4ur0n/code/haskell-language-server/ghcide/src/Development/IDE/Core/OfInterest.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:  Failed to parse result of calling cabalfdLock: invalid argument (Invalid argument)
Left [CradleError ExitSuccess ["Multi Cradle: No prefixes matched","pwd: /home/sir4ur0n/code/haskell-language-server","filepath: /home/sir4ur0n/code/haskell-language-server/test/testdata/HaReRename.hs","prefixes:","(\"./test/functional/\",Stack {component = Just \"haskell-language-server:func-test\"})","(\"./test/utils/\",Stack {component = Just \"haskell-language-server:func-test\"})","(\"./exe/Main.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server\"})","(\"./exe/Wrapper.hs\",Stack {component = Just \"haskell-language-server:exe:haskell-language-server-wrapper\"})","(\"./src\",Stack {component = Just \"haskell-language-server:lib\"})","(\"./ghcide/src\",Stack {component = Just \"ghcide:lib:ghcide\"})","(\"./ghcide/exe\",Stack {component = Just \"ghcide:exe:ghcide\"})"]]
File:     /home/sir4ur0n/code/haskell-language-server/test/testdata/HaReRename.hs
Hidden:   no
Range:    1:0-100001:0
Consulting the cradle for "/home/sir4ur0n/code/haskell-language-server/test/functional/Diagnostic.hs"
File:     /home/sir4ur0n/code/haskell-language-server/test/testdata/HaReRename.hs
Hidden:   no
Range:    1:0-100001:0
Source:   compiler
Severity: DsError
Message:
  Multi Cradle: No prefixes matched
  pwd: /home/sir4ur0n/code/haskell-language-server
  filepath: /home/sir4ur0n/code/haskell-language-server/test/testdata/HaReRename.hs
  prefixes:
  ("./test/functional/",Stack {component = Just "haskell-language-server:func-test"})
  ("./test/utils/",Stack {component = Just "haskell-language-server:func-test"})
  ("./exe/Main.hs",Stack {component = Just "haskell-language-server:exe:haskell-language-server"})
  ("./exe/Wrapper.hs",Stack {component = Just
  "haskell-language-server:exe:haskell-language-server-wrapper"})
  ("./src",Stack {component = Just "haskell-language-server:lib"})
  ("./ghcide/src",Stack {component = Just "ghcide:lib:ghcide"})
  ("./ghcide/exe",Stack {component = Just "ghcide:exe:ghcide"})
> Using main module: 1. Package `haskell-language-server' component haskell-language-server:test:func-test with main-is file: /home/sir4ur0n/code/haskell-language-server/test/functional/Main.hs
> haskell-language-server> configure (lib + exe)
> Configuring haskell-language-server-0.1.0.0...
> haskell-language-server> initial-build-steps (lib + exe)
> The following GHC options are incompatible with GHCi and have not been passed to it: -threaded
> Configuring GHCi with the following packages: haskell-language-server
>
> Warning: Didn't find expected autogen file:
>          /home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test/autogen/cabal_macros.h
> /home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb:/home/sir4ur0n/.stack/snapshots/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb:/home/sir4ur0n/.stack/programs/x86_64-linux/ghc-tinfo6-8.6.5/lib/ghc-8.6.5/package.conf.d
Right (ComponentOptions {componentOptions = ["-i","-odir=/home/sir4ur0n/code/haskell-language-server/.stack-work/odir","-hidir=/home/sir4ur0n/code/haskell-language-server/.stack-work/odir","-hide-all-packages","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test","-i/home/sir4ur0n/code/haskell-language-server/test/utils","-i/home/sir4ur0n/code/haskell-language-server/test/functional","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test/autogen","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/global-autogen","-i/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build/func-test/func-test-tmp","-stubdir=/home/sir4ur0n/code/haskell-language-server/.stack-work/dist/x86_64-linux-tinfo6/Cabal-2.4.0.1/build","-package-id=base-4.12.0.0","-package=haskell-language-server-0.1.0.0","-package-id=haskell-lsp-0.22.0.0-Dm3Y5tmF7yh4ECsA5etaU6","-package-id=hie-bios-0.5.0-FSBb2ut8sK4IXHsh2geoHJ","-package-id=aeson-1.4.6.0-4uBfdXbzscjDeNqrTRfhWo","-package-id=blaze-markup-0.8.2.3-ATcS0IRQcDMCxhB8vW9N1d","-package-id=containers-0.6.0.1","-package-id=data-default-0.7.1.1-COovZVyOTYqEavTGLlfqy8","-package-id=directory-1.3.3.0","-package-id=filepath-1.4.2.1","-package-id=hslogger-1.2.12-EOh2s68fQfBFyuAt5sDs18","-package-id=hspec-2.7.1-28Bzukly5w86FpiwMyuAYC","-package-id=hspec-core-2.7.1-IbyPArxIL6526nHHCZLmW2","-package-id=lsp-test-0.10.3.0-8UmPD9MvFJRH3G2kYk9Unq","-package-id=stm-2.5.0.0","-package-id=tasty-hunit-0.10.0.2-FgUrIbgCsUcDDS8XJ0iyJm","-package-id=text-1.2.3.1","-package-id=unordered-containers-0.2.10.0-LgoTL3wbBEY5bZIDJiyxW4","-package-id=yaml-0.11.2.0-8COcGy2e3i5ABQkqGF0U1e","-package-id=haskell-lsp-types-0.22.0.0-EqCdNjFLrl1H6FLGsAPz20","-package-id=hspec-expectations-0.8.2-xEhsaJyqHC4OUn6v2CIJJ","-package-id=lens-4.18-HbPjKqYp8F3CoEHux0jQ2","-package-id=tasty-1.2.3-GiudK7yenNPC8Gtev28Pwr","-package=tasty-ant-xml-1.1.6","-package=tasty-expected-failure-0.11.1.2","-package=tasty-rerun-1.1.17","-Wall","-Wredundant-constraints","-Wall","-Wno-name-shadowing","-rtsopts","-with-rtsopts=-N","-DAGPL","-DAGPL","-ghci-script=/tmp/haskell-stack-ghci/e65fba7d/ghci-script","-package-db","/home/sir4ur0n/code/haskell-language-server/.stack-work/install/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb","-package-db","/home/sir4ur0n/.stack/snapshots/x86_64-linux-tinfo6/ad782cb2674a8c9bca215819570aaed13a825613b404f3308cb64df7b909b5f5/8.6.5/pkgdb","-package-db","/home/sir4ur0n/.stack/programs/x86_64-linux/ghc-tinfo6-8.6.5/lib/ghc-8.6.5/package.conf.d"], componentRoot = "/home/sir4ur0n/code/haskell-language-server", componentDependencies = ["haskell-language-server.cabal","package.yaml","stack.yaml"]})
haskell-language-server: <command line>: cannot satisfy -package haskell-language-server-0.1.0.0
    (use -v for more information)
```

Besides the PR I just opened, I **still need to run `stack test` in order to have a working HLS**. Otherwise I still get `haskell-language-server: <command line>: cannot satisfy -package haskell-language-server-0.1.0.0` and hovering remains stuck on `Loading...`.

So there is another issue to fix.

Once I run `stack test` (or just `stack build` but then HLS doesn't work for test components) it works much better.

Note, I still get 3 errors in 2 `test/functional` modules, which seem related to the parser breaking on `-- $` chain of characters. I will open another issue to track this.

```
/home/sir4ur0n/code/haskell-language-server/test/functional/Deferred.hs:162:9: error:
    parse error on input ‘-- $ runSession hieCommand fullCaps "test/testdata" $ do’parser
```

--- a/README.md
+++ b/README.md
@@ -497,15 +497,27 @@ args = ["--lsp"]
 
 ### Hacking on haskell-language-server
 
-Haskell-language-server can be used on its own project.  We have supplied
-preset samples of `hie.yaml` files for stack and cabal, simply copy
-the appropriate template to `hie.yaml` and it should work.
+Haskell-language-server can be used on itself. We provide
+preset samples of `hie.yaml` for Cabal and Stack.
 
-- `hie.yaml.cbl` for cabal
-- `hie.yaml.stack` for stack
+Note: the `./install/` folder is not directly tied to the project so it has dedicated `./install/hie.yaml.[cbl|stack]` 
+templates.
 
-Two sample `hie.yaml` files are provided, `hie.yaml.stack` for stack
-usage, `hie.yaml.cbl` for cabal. Simply copy the relevant one to be
-`hie.yaml` and it should work.
+#### Using Cabal
 
-The developers tend to hang out at [our IRC channel](https://webchat.freenode.net/?channels=haskell-ide-engine) at `#haskell-ide-engine` on `freenode`.
+```shell
+$ cp hie.yaml.cbl hie.yaml
+$ cp install/hie.yaml.cbl install/hie.yaml
+```
+
+#### Using Stack
+
+Note: Stack project must also be built once until [this issue](https://github.com/commercialhaskell/stack/issues/5213) is fixed.
+
+```shell
+$ cp hie.yaml.stack hie.yaml
+$ cp install/hie.yaml.stack install/hie.yaml
+$ stack build --test --no-run-tests
+$ cd install
+$ stack build
+```

--- a/hie.yaml.cbl
+++ b/hie.yaml.cbl
@@ -2,25 +2,40 @@
 # in hie, using cabal as the build system.  To use is, copy it to a
 # file called 'hie.yaml'
 cradle:
-  cabal:
+  multi:
+    - path: "./test/testdata/"
+      config: { cradle: { none:  } }
 
-    - path: "./test/functional/"
-      component: "haskell-language-server:func-test"
+    - path: "./install/"
+      config: { cradle: { none:  } }
 
-    - path: "./test/utils/"
-      component: "haskell-language-server:func-test"
+    - path: "./"
+      config:
+        cradle:
+          cabal:
+            - path: "./test/functional/"
+              component: "haskell-language-server:func-test"
 
-    - path: "./exe/Main.hs"
-      component: "haskell-language-server:exe:haskell-language-server"
+            - path: "./test/utils/"
+              component: "haskell-language-server:func-test"
+        
+            - path: "./exe/Main.hs"
+              component: "haskell-language-server:exe:haskell-language-server"
+        
+            - path: "./exe/Arguments.hs"
+              component: "haskell-language-server:exe:haskell-language-server"
 
-    - path: "./exe/Wrapper.hs"
-      component: "haskell-language-server:exe:haskell-language-server-wrapper"
+            - path: "./exe/Wrapper.hs"
+              component: "haskell-language-server:exe:haskell-language-server-wrapper"
+        
+            - path: "./src"
+              component: "lib:haskell-language-server"
 
-    - path: "./src"
-      component: "lib:haskell-language-server"
-
-    - path: "./ghcide/src"
-      component: "ghcide:lib:ghcide"
-
-    - path: "./ghcide/exe"
-      component: "ghcide:exe:ghcide"
+            - path: "./.stack-work/"
+              component: "lib:haskell-language-server"
+        
+            - path: "./ghcide/src"
+              component: "ghcide:lib:ghcide"
+        
+            - path: "./ghcide/exe"
+              component: "ghcide:exe:ghcide"

--- a/hie.yaml.cbl
+++ b/hie.yaml.cbl
@@ -6,9 +6,6 @@ cradle:
     - path: "./test/testdata/"
       config: { cradle: { none:  } }
 
-    - path: "./install/"
-      config: { cradle: { none:  } }
-
     - path: "./"
       config:
         cradle:

--- a/hie.yaml.stack
+++ b/hie.yaml.stack
@@ -6,9 +6,6 @@ cradle:
     - path: "./test/testdata/"
       config: { cradle: { none:  } }
 
-    - path: "./install/"
-      config: { cradle: { none:  } }
-
     - path: "./"
       config:
         cradle:

--- a/hie.yaml.stack
+++ b/hie.yaml.stack
@@ -2,24 +2,40 @@
 # in hie, using stack as the build system.  To use is, copy it to a
 # file called 'hie.yaml'
 cradle:
-  stack:
-    - path: "./test/functional/"
-      component: "haskell-language-server:func-test"
+  multi:
+    - path: "./test/testdata/"
+      config: { cradle: { none:  } }
 
-    - path: "./test/utils/"
-      component: "haskell-language-server:func-test"
+    - path: "./install/"
+      config: { cradle: { none:  } }
 
-    - path: "./exe/Main.hs"
-      component: "haskell-language-server:exe:haskell-language-server"
+    - path: "./"
+      config:
+        cradle:
+          stack:
+            - path: "./test/functional/"
+              component: "haskell-language-server:func-test"
 
-    - path: "./exe/Wrapper.hs"
-      component: "haskell-language-server:exe:haskell-language-server-wrapper"
+            - path: "./test/utils/"
+              component: "haskell-language-server:func-test"
 
-    - path: "./src"
-      component: "haskell-language-server:lib"
+            - path: "./exe/Main.hs"
+              component: "haskell-language-server:exe:haskell-language-server"
+        
+            - path: "./exe/Arguments.hs"
+              component: "haskell-language-server:exe:haskell-language-server"
 
-    - path: "./ghcide/src"
-      component: "ghcide:lib:ghcide"
+            - path: "./exe/Wrapper.hs"
+              component: "haskell-language-server:exe:haskell-language-server-wrapper"
+        
+            - path: "./src"
+              component: "haskell-language-server:lib"
 
-    - path: "./ghcide/exe"
-      component: "ghcide:exe:ghcide"
+            - path: "./.stack-work/"
+              component: "haskell-language-server:lib"
+        
+            - path: "./ghcide/src"
+              component: "ghcide:lib:ghcide"
+        
+            - path: "./ghcide/exe"
+              component: "ghcide:exe:ghcide"

--- a/install/hie.yaml.cbl
+++ b/install/hie.yaml.cbl
@@ -1,0 +1,3 @@
+cradle: 
+  cabal: 
+    component: "lib:hls-install"

--- a/install/hie.yaml.stack
+++ b/install/hie.yaml.stack
@@ -1,0 +1,3 @@
+cradle: 
+  stack: 
+    component: "hls-install:lib"

--- a/install/stack.yaml
+++ b/install/stack.yaml
@@ -1,0 +1,13 @@
+resolver: lts-14.27 # Last 8.6.5 
+
+packages:
+  - .
+
+extra-deps:
+  - cabal-install-parsers-0.3.0.1
+  - Cabal-3.2.0.0
+  - binary-instances-1.0.0.1
+  - lukko-0.1.1.2
+
+nix:
+  packages: [ icu libcxx zlib ]

